packages/page-claims/src/index.tsx

```diff
@@ -73,183 +68,242 @@ const Signature = styled.textarea`
-                    />
-                  </Button.Group>
-                )}
-              </Card>
-            )}
-          </Column>
-          <Column showEmptyText={false}>
-            {(step >= Step.Claim) && (
-              <ClaimDisplay
-                button={this.renderTxButton()}
-                ethereumAddress={ethereumAddress}
-              />
-            )}
-          </Column>
-        </Columar>
-      </main>
-    );
-  }
+  const goToStepSign = useCallback(() => {
+    setStep(Step.Sign);
+  }, []);
 
-  protected isDisabled = (): boolean => {
-    const { accountId, signature } = this.state;
+  const goToStepClaim = useCallback(() => {
+    setStep(Step.Claim);
+  }, []);
 
-    return !accountId || !signature;
-  }
+  // Depending on the account, decide which step to show.
+  const handleAccountStep = useCallback(() => {
+    if (isPreclaimed) {
+      goToStepClaim();
+    } else if (ethereumAddress || isOldClaimProcess) {
+      goToStepSign();
+    } else {
+      setStep(Step.ETHAddress);
+    }
+  }, [ethereumAddress, goToStepClaim, goToStepSign, isPreclaimed, isOldClaimProcess]);
 
-  protected isUnsigned = (): boolean => true;
+  const onChangeSignature = useCallback((event: React.SyntheticEvent<Element>) => {
+    const { value: signatureJson } = event.target as HTMLInputElement;
 
-  protected submitLabel = (): React.ReactNode => this.props.t('Redeem');
+    const { ethereumAddress, signature } = recoverFromJSON(signatureJson);
 
-  protected txMethod = (): string => 'claims.claim';
+    setEthereumAddress(ethereumAddress?.toString());
+    setSignature(signature);
+  }, []);
 
-  protected txParams = (): [string | null, EcdsaSignature | null] => {
-    const { accountId, signature } = this.state;
+  const onChangeEthereumAddress = useCallback((value: string) => {
+    // FIXME We surely need a better check than just a trim
 
-    return [
-      accountId ? accountId.toString() : null,
-      signature || null
-    ];
-  }
+    setEthereumAddress(value.trim());
+  }, []);
 
-  protected onChangeAccount = (accountId: string | null): void => {
-    this.setState(({ step }: State): Pick<State, never> => {
-      return {
-        ...(
-          step > Step.Account
-            ? this.defaultState
-            : {}
-        ),
-        accountId
-      };
-    });
-  }
+  const onCopy = useCallback(() => {
+    setDidCopy(true);
+  }, []);
 
-  protected onChangeSignature = (event: React.SyntheticEvent<Element>): void => {
-    const { value: signatureJson } = event.target as HTMLInputElement;
+  // If it's 1/ not preclaimed and 2/ not the old claiming process, fetch the
+  // statement kind to sign.
+  const _ethereumAddress = ethereumAddress ? api.createType('EthereumAddress', ethereumAddress) : null;
+  const statementKind = useCall<StatementKind | null>(!isPreclaimed && !isOldClaimProcess && _ethereumAddress && api.query.claims.signing, [_ethereumAddress], {
```

Comment:
You don't want to pass a value that changes every render into `useCall` and params. (Well, it does work, but is smelly)